odoo/addons/base/views/ir_qweb_widget_templates.xml

```diff
@@ -33,6 +33,20 @@
                 <span>
                     <div>
                         <span itemprop="addressLocality" t-esc="city"/>,
+                        <t t-if="object.state_id and object.country_id.state_required">
```

Comment:
Actually we could probably compute variables like "has_city" and "has_state_id" first and use it in various code blocks :)

Reply:
Hello @tde-banana-odoo :slightly_smiling_face: 
Is it mean something like this?
As it seems good and we don't need to pass the `country_id` also explicitly in the fields while using the widget with this approach.
```xml
            <t t-set="has_city" t-value="city"/>
            <t t-set="has_state" t-value="object.state_id and object.country_id.state_required"/>
            <t t-set="has_country" t-value="country_id"/>
            <t t-set="location" t-value="[]"/>
            <t t-if="has_city" t-set="location" t-value="location + [city]"/>
            <t t-if="has_state" t-set="location" t-value="location + [object.state_id.code]"/>
            <t t-if="has_country" t-set="location" t-value="location + [country_id]"/>```
            
            Or am I missing something?

Reply:
Including the "fname in fields" part, like
```
    <t t-set="has_city" t-value="city and 'city' in fields"/>
    <t t-set="has_state" t-value="object.state_id and 'state_id' in fields"/>
    <t t-set="has_country" t-value="country_id and 'country_id' in fields "/>
```